### ISSUE TITLE
Cannot find name 'URLPattern'

### ISSUE THREAD

**Issue by lt-hacker** [denoland/fresh#76]
This error occurs intermittently when running my server:

```
Download https://deno.land/std@0.95.0/_util/assert.ts
Download https://cdn.esm.sh/v56/preact@10.5.15/src/index.d.ts
Download https://cdn.esm.sh/v56/preact@10.5.15/src/jsx.d.ts
Download https://crux.land/api/get/2KNRVU.ts
Check file:///Users/quenio/Projects/LiteTrade/indiepro/main.ts
error: TS2304 [ERROR]: Cannot find name 'URLPattern'.
        const pattern = new URLPattern({
                            ~~~~~~~~~~
    at https://crux.land/api/get/2KNRVU.ts:136:29
Watcher Process finished. Restarting on file change...
```

My `deps_client.ts`:

```
/// <reference no-default-lib="true" />
/// <reference lib="dom" />
/// <reference lib="dom.asynciterable" />
/// <reference lib="deno.ns" />
/// <reference lib="deno.unstable" />

export * from "https://raw.githubusercontent.com/lucacasonato/fresh/main/runtime.ts"
```

My `deps_server.ts`:
```
export * from "https://raw.githubusercontent.com/lucacasonato/fresh/main/server.ts"
```

**Comment by tsutorm** [comment]
Hello @quenio-lt

I'm having trouble with the same symptom, 
but I've avoided it by adding the `--no-check` option to the execution command.

```
deno run -A --watch --no-check main.ts
```

I think there is a problem with the type checking mechanism of the deno runtime.

**Comment by sylc** [comment]
I think it is related to https://github.com/denoland/deno/issues/12558

Another work around, instead of disabling ts entirely with `--no-check`, you can declare it

add the below to deps.ts
```
// temporary workaround until URLPattern is defined
declare global {
  // deno-lint-ignore no-explicit-any no-var
  var URLPattern: any;
}
```

**Comment by lucacasonato** [comment]
It is indeed caused by denoland/deno#12558.

**Comment by lucacasonato** [comment]
This is fixed in Deno [v1.21.2](https://github.com/denoland/deno/releases/tag/v1.21.2) and up, when using fresh commit https://github.com/lucacasonato/fresh/commit/c1e376a3623dbd352ba0b479213dbc3535844f74 or later.
